components/director/internal/domain/api/fixtures_test.go

```diff
@@ -0,0 +1,344 @@
+		Value:           "1.0.0",
+		Deprecated:      &deprecated,
+		DeprecatedSince: &deprecatedSince,
+		ForRemoval:      &forRemoval,
+	}
+
+	auth1 := model.Auth{
+		Credential: model.CredentialData{
+			Basic: &model.BasicCredentialData{
+				Username: "test",
+				Password: "pwd",
+			},
+		},
+		AdditionalHeaders:     map[string][]string{"testHeader": {"hval1", "hval2"}},
+		AdditionalQueryParams: map[string][]string{"testParam": {"pval1", "pval2"}},
+		RequestAuth:           &model.CredentialRequestAuth{},
+	}
+	auth2 := model.Auth{
+		Credential: model.CredentialData{
+			Basic: &model.BasicCredentialData{
+				Username: "test",
+				Password: "pwd",
+			},
+		},
+		AdditionalHeaders:     map[string][]string{"testHeader": {"hval1", "hval2"}},
+		AdditionalQueryParams: map[string][]string{"testParam": {"pval1", "pval2"}},
+		RequestAuth:           &model.CredentialRequestAuth{},
+	}
+
+	runtimeAuth1 := model.RuntimeAuth{
+		RuntimeID: "1",
+		Auth:      &auth1,
+	}
+	runtimeAuth2 := model.RuntimeAuth{
+		RuntimeID: "2",
+		Auth:      &auth2,
+	}
+
+	return &model.APIDefinition{
+		ID:            id,
+		ApplicationID: "applicationID",
+		Name:          name,
+		Description:   &description,
+		Spec:          spec,
+		TargetURL:     "https://test-url.com",
+		Group:         &group,
+		Auth:          &runtimeAuth1,                                      //TODO: https://github.com/kyma-incubator/compass/issues/67
+		Auths:         []*model.RuntimeAuth{&runtimeAuth1, &runtimeAuth2}, //TODO: https://github.com/kyma-incubator/compass/issues/67
+		DefaultAuth:   &auth1,                                             //TODO: https://github.com/kyma-incubator/compass/issues/67
+		Version:       version,
+	}
+}
+
+func fixDetailedGQLApiDefinition(t *testing.T, id, name, description string, group string) *graphql.APIDefinition {
+	time, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	data := graphql.CLOB("data")
+	filter := "filter"
+	format := graphql.SpecFormatJSON
+
+	spec := &graphql.APISpec{
+		Data:   &data,
+		Format: &format,
+		Type:   graphql.APISpecTypeOpenAPI,
+		FetchRequest: &graphql.FetchRequest{
+			URL:    "https://test-fetch-request.com",
+			Auth:   nil,
+			Mode:   graphql.FetchModeSingle,
+			Filter: &filter,
+			Status: &graphql.FetchRequestStatus{
+				Condition: graphql.FetchRequestStatusConditionInitial,
+				Timestamp: graphql.Timestamp(time),
+			},
+		},
+	}
+
+	deprecated := false
+	deprecatedSince := ""
+	forRemoval := false
+
+	version := &graphql.Version{
+		Value:           "1.0.0",
+		Deprecated:      &deprecated,
+		DeprecatedSince: &deprecatedSince,
+		ForRemoval:      &forRemoval,
+	}
+
+	headers1 := graphql.HttpHeaders{"testHeader": {"hval1", "hval2"}}
+	headers2 := graphql.HttpHeaders{"testHeader": {"hval1", "hval2"}}
+	params1 := graphql.QueryParams{"testParam": {"pval1", "pval2"}}
+	params2 := graphql.QueryParams{"testParam": {"pval1", "pval2"}}
+
+	auth1 := graphql.Auth{
```

Comment:
BTW Don't pass all values for Auth, as the Auth converter is tested somewhere else. What you need to do is to test only API converter and that's it. You're mocking Auth converter anyway, so please decrease complexity here.